### PR TITLE
UC-143 | remove private wikis from user favorite wikis

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/UC-143

### CODE REVIEW

File: extensions/wikia/UserProfilePageV3/filters/UserWikisFilterPrivateDecorator.class.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * Class UserWikisFilterPrivateDecorator
+ */
+class UserWikisFilterPrivateDecorator extends UserWikisFilterDecorator {
+	public function getFiltered() {
+		$filtered = $this->filter->getFiltered();
+
+		foreach( $filtered as $key => $wikiData ) {
+			$wikiId = (int) $wikiData['id'];
+
+			if( WikiFactory::getVarValueByName( 'wgIsPrivateWiki', $wikiId ) === true ) {
```
Comment (armonr) [inline extensions/wikia/UserProfilePageV3/filters/UserWikisFilterPrivateDecorator.class.php]: Minor: comparison is not needed.

File: extensions/wikia/UserProfilePageV3/filters/UserWikisFilterPrivateDecorator.class.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * Filters private wikis from user favourite wikis list
+ *
+ * Class UserWikisFilterPrivateDecorator
+ */
+class UserWikisFilterPrivateDecorator extends UserWikisFilterDecorator {
+	public function getFiltered() {
+		$filtered = $this->filter->getFiltered();
+
+		foreach( $filtered as $key => $wikiData ) {
+			$wikiId = (int) $wikiData['id'];
```
Comment (armonr) [inline extensions/wikia/UserProfilePageV3/filters/UserWikisFilterPrivateDecorator.class.php]: Spacing: after `foreach`, `if`, and before/after `int`

**Comment by armonr** [comment]
Your change set look good. However, instead of adding another filter, it may be better to improve these filters. At first glance, I did not find any other usages of these filters other than that one place.
It may be safe to label these filter classes as [YAGNI](http://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) unless there may be a reason we might want to use only one of the filters somewhere. Otherwise, at the current state, it is quite inefficient to run through a list of wikis for each filter.
So I suggest a refactoring of this code :)

File: extensions/wikia/UserProfilePageV3/filters/UserWikisFilterPrivateDecorator.class.php (hunk 1 above)
Comment (kvas-damian) [inline extensions/wikia/UserProfilePageV3/filters/UserWikisFilterPrivateDecorator.class.php]: Nice catch!
Overused Zen of Pyhon:
> [...]
> Explicit is better than implicit.
> [...]

But in this case it's definitely not required :)

**Comment by kvas-damian** [comment]
@armonr Good point, My feeling was almost the same while touching this code, however I decided to not combine filters. This way [Single responsibility principle](http://en.wikipedia.org/wiki/Single_responsibility_principle) is still implemented here.

Additionally I considered performance implications of this decision and found that list computed by those filters has limited size - this time cost/benefit ration didn't convince. https://github.com/Wikia/app/blob/UC-143/extensions/wikia/UserProfilePageV3/models/FavoriteWikisModel.class.php#L12

If you still feel that we should improve this code, please ping me I'll change it.

**Comment by armonr** [comment]
@kvas-damian You also have a good cost analysis case :)
:shipit:

**Comment by lizlux** [comment]
LGTM

**Pull request merged**